packages/rocketchat-statistics/server/lib/SAUMonitor.js

```diff
@@ -0,0 +1,306 @@
+		this._instanceId = null;
+	}
+
+	start(instanceId) {
+		if (this.isRunning()) {
+			return;
+		}
+
+		this._instanceId = instanceId;
+
+		if (!this._instanceId) {
+			logger.debug('[start] - InstanceId is not defined.');
+			this._log(`${ this._serviceName } - Error starting monitor: InstanceID is not defined.`);
+			return;
+		}
+
+		this._startMonitoring(() => {
+			this._started = true;
+			this._log(`${ this._serviceName } - Started: ${ this._instanceId }`);
+			logger.debug(`[start] - InstanceId: ${ this._instanceId }`);
+		});
+	}
+
+	stop() {
+		if (!this.isRunning()) {
+			return;
+		}
+
+		this._started = false;
+
+		if (this._timer) {
+			Meteor.clearInterval(this._timer);
+		}
+
+		this.log(`${ this._serviceName } - Stopped.`);
+		logger.debug(`[stop] - InstanceId: ${ this._instanceId }`);
+	}
+
+	isRunning() {
+		return typeof this._started === 'boolean' && this._started === true;
+	}
+
+	_log(...args) {
+		if (this._debug === true) {
+			console.log.apply(console, args);
+		}
+	}
+
+	setDebug(mode) {
+		if (typeof mode !== 'boolean') {
+			return;
+		}
+
+		this._debug = mode;
+	}
+
+	async _startMonitoring(callback) {
+		try {
+			await this._handleAccountEvents();
+			await this._handleOnConnection();
+			await this._startSessionControl();
+			await this._initActiveServerSessions();
+			if (callback) {
+				callback();
+			}
+		} catch (err) {
+			throw new Meteor.Error(err);
+		}
+	}
+
+	_startSessionControl() {
+		if (this.isRunning()) {
+			return;
+		}
+
+		if (this._monitorTime < 0) {
+			return;
+		}
+
+		this._timer = Meteor.setInterval(() => {
+			this._updateActiveSessions();
+		}, this._monitorTime);
+	}
+
+	_handleOnConnection() {
+		if (this.isRunning()) {
+			return;
+		}
+
+		Meteor.onConnection((connection) => {
+			this._handleSession(connection, getDateObj());
+
+			connection.onClose(() => {
+				RocketChat.models.Sessions.closeByInstanceIdAndSessionId(this._instanceId, connection.id);
+			});
+		});
+	}
+
+	_handleAccountEvents() {
+		if (this.isRunning()) {
+			return;
+		}
+
+		Accounts.onLogin((info) => {
+			const userId = info.user._id;
+			const loginAt = new Date();
+			const params = { userId, loginAt, ...getDateObj() };
+			this._handleSession(info.connection, params);
+			this._updateConnectionInfo(info.connection.id, { loginAt });
+		});
+
+		Accounts.onLogout((info) => {
+			const sessionId = info.connection.id;
+			const userId = info.user._id;
+			RocketChat.models.Sessions.logoutByInstanceIdAndSessionIdAndUserId(this._instanceId, sessionId, userId);
+		});
+	}
+
+	_handleSession(connection, params) {
+		const data = this._getConnectionInfo(connection, params);
+		RocketChat.models.Sessions.createOrUpdate(data);
+	}
+
+	_updateActiveSessions() {
+		if (!this.isRunning()) {
+			return;
+		}
+
+		const { year, month, day } = this._today;
+		const currentDateTime = new Date();
+		const currentDay = getDateObj(currentDateTime);
+
+		if (JSON.stringify(this._today) !== JSON.stringify(currentDay)) {
```

Comment:
What happens if the json properties order are different?

Reply:
In the both variables their values come from the same function, so it would be really weird if that happens, but I can guarantee the props order or change this approach if it's needed.
I'll improve this approach and push a new commit.